code/datums/elements/hat_wearer.dm

```diff
@@ -0,0 +1,73 @@
+/**
+ * # mobs that can wear hats!
+ */
+/datum/element/hat_wearer
+	element_flags = ELEMENT_DETACH_ON_HOST_DESTROY|ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	///offsets of hats we will wear
+	var/list/offsets
```

Comment:
Not related to above but it is a great shame that cyborgs do not use this element.